webknossos/webknossos/dataset/dataset.py

```diff
@@ -1339,6 +1352,62 @@ def add_layer_from_images(
                     "[WARNING] Some images are larger than expected, smaller slices are padded with zeros now. "
                     + f"New bbox is {layer.bounding_box}, expected {expected_bbox}."
                 )
+
+            # Check if category of layer is set correctly
+            try:
+                if not user_set_category:
+                    guessed_category = guess_category_from_view(layer.get_finest_mag())
+                    if guessed_category != layer.category:
+                        if guessed_category == SEGMENTATION_CATEGORY:
+                            logging.info("The layer category is set to segmentation.")
+                            segmentation_layer_properties: SegmentationLayerProperties = SegmentationLayerProperties(
+                                **(
+                                    attr.asdict(layer._properties, recurse=False)
+                                ),  # use all attributes from LayerProperties
+                                largest_segment_id=int(max(max_ids)),
+                            )
+                            segmentation_layer_properties.category = (
+                                SEGMENTATION_CATEGORY
+                            )
+                            self._properties.data_layers = [
+                                (
+                                    layer_properties
+                                    if layer_properties.name != layer.name
+                                    else segmentation_layer_properties
+                                )
+                                for layer_properties in self._properties.data_layers
+                            ]
+                            (self.path / layer_name).mkdir(parents=True, exist_ok=True)
+                            self._layers[layer_name] = SegmentationLayer(
+                                self, segmentation_layer_properties
+                            )
+                            self._export_as_json()
```

Comment:
I hope this can be DRYed a bit with the else-branch. Also, maybe the properties should provide a `updateForLayer` method which makes it easier to update one specific layer? Assigning to `_properties.data_layers` looks a bit verbose.

Reply:
I added an `updateForLayer` method and moved parts that are in both branches to the end of both.